### ISSUE TITLE
WebM video rejected with “Video has no video stream”

### ISSUE THREAD

**Issue by AndrewKvalheim** [mastodon/mastodon#19172]
### Steps to reproduce the problem

Attach this WebM video to a post:

[catalogchoice.webm](https://user-images.githubusercontent.com/1844746/189794871-4f583e02-c413-4a6a-b254-d39d99b9f373.webm)

### Expected behaviour

Inclusion in the post as a playable video, or a message explaining that the format is unsupported

### Actual behaviour

“422 Video has no video stream” is displayed

### Specifications

Server: Mastodon v3.5.1 on Ubuntu 18.04.6 LTS  
Client: Web interface in Firefox 104.0.2 and Chromium 105.0.5195.102

**Comment by trwnh** [comment]
which client app are you using? i had a similar issue that i later identified to be in SubwayTooter but not in the web app

**Comment by AndrewKvalheim** [comment]
Web interface via Firefox 104.0.2 and Chromium 105.0.5195.102

**Comment by ClearlyClaire** [comment]
What I think may be happening is that this video has a size (1645x1042) that would be invalid with the format and codecs Mastodon re-encodes it to.

**Comment by CleoMenezesJr** [comment]
Same thing happens to me. I'm using Mastodon v4.0.2.
This is the video I'm trying to post:
[Screencast from 2022-11-27 14-05-16.webm](https://user-images.githubusercontent.com/54215258/204149958-d8c04e6d-a84f-4e7b-a50d-e6d40ddad69c.webm)

**Comment by aral** [comment]
[Screencast from 2023-02-24 16-54-32.webm](https://user-images.githubusercontent.com/41798/221243252-5f95a7fd-9f30-4184-87bb-adc6a5f7a5cc.webm)

Same issue with this one.

**Comment by Freeplayg** [comment]
I use another Fedi platform that does support Webm, however when viewed from Mastodon it does get this error.

**Comment by cachapa** [comment]
Many of the people here appear to have stumbled upon this issue because they were trying to post a Gnome screen recording (as I have).

As a workaround it's possible to convert the webm file to mp4 using ffmpeg, though that's not trivial since the video needs to have an even width and height, and for some reason the output has 1000 fps (???).

Anyway, I wrote a shell script that crops the video by 1 px if needed, and forces the framerate to 60fps:

```shell
#/bin/bash

OUT="$(basename -- "$1" .webm).mp4"

ffmpeg -i "$1" -vf "crop=floor(iw/2)*2:floor(ih/2)*2" -r 60 "$OUT"
```

**Comment by ClearlyClaire** [comment]
I believe this should be fixed by #26500

**Comment by AndrewKvalheim** [comment]
Thanks, this video fits the description—

```console
$ ffprobe -i 'catalogchoice.webm' -show_streams | grep 'frame_rate'
r_frame_rate=60000/1001
avg_frame_rate=0/0
```

—and I’ve confirmed that applying #26500 allows successful attachment. So this is actually a duplicate of #14779 which was closed prematurely.
